### PR TITLE
Improved Class Level Docstrings for GLMs

### PR DESCRIPTION
**Pull request by BalzaniEdoardo**
## Updates
                                                                                                                                                                                     
### GLM (glm.py):                                                                                                                                                                             
  - Fit a GLM - Basic model fitting example                                                                                                                                                 
  - Customize the Observation Model - String and object observation model                                                                                                                   
  - Customize the Inverse Link Function - Using softplus                                                                                                                                    
  - Use Regularization - Ridge and Lasso examples                                                                                                                                           
  - Select a Solver - LBFGS example                                                                                                                                                         
                                                                                                                                                                                            
### PopulationGLM (glm.py):                                                                                                                                                                   
  - Fit a PopulationGLM - Basic population model fitting                                                                                                                                    
  - Mask Coefficients with an Array - Array-based feature mask                                                                                                                              
  - Mask Coefficients with a FeaturePytree - FeaturePytree-based mask                                                                                                                       
  - Customize the Observation Model - Gamma observation model                                                                                                                               
  - Use Regularization - Ridge regularization example                                                                                                                                       
                                                                                                                                                                                            
### ClassifierGLM (classifier_glm.py):                                                                                                                                                        
  - Fit a ClassifierGLM - Basic binary classification                                                                                                                                       
  - Predict Class Labels - Getting predictions                                                                                                                                              
  - Predict Class Probabilities - Proba and log-proba                                                                                                                                       
  - Use String Labels - Using string class labels like "cat"/"dog"                                                                                                                          
  - Multi-class Classification - 3-class example                                                                                                                                            
  - Use Regularization - Ridge with custom strength                                                                                                                                         
                                                                                                                                                                                            
### ClassifierPopulationGLM (classifier_glm.py):                                                                                                                                              
  - Fit a ClassifierPopulationGLM - Basic population classifier                                                                                                                             
  - Predict Class Labels - Getting predictions for each neuron                                                                                                                              
  - Predict Class Probabilities - Probabilities per neuron                                                                                                                                  
  - Use String Labels - Using string labels                                                                                                                                                 
  - Use a Feature Mask - 3D feature mask example                                                                                                                                            
  - Use Regularization - Ridge with custom strength

### CODE REVIEW

File: src/nemos/glm/classifier_glm.py [hunk 1]
```diff
@@ -908,14 +963,70 @@ class labels from the provided ``y``. If you set ``coef_`` and ``intercept_`` ma
 
     Examples
     --------
+    **Fit a ClassifierPopulationGLM**
+
+    Basic multi-class classification for a population of neurons:
```
Comment (gviejo) [inline src/nemos/glm/classifier_glm.py]: What does it mean here?

Comment (gviejo) [inline src/nemos/glm/classifier_glm.py]:
```suggestion
    Basic multi-class classification for multi-subjects
```

File: src/nemos/glm/classifier_glm.py [hunk 2]
```diff
@@ -908,14 +963,70 @@ class labels from the provided ``y``. If you set ``coef_`` and ``intercept_`` ma
...
-    >>> model = nmo.glm.ClassifierPopulationGLM(n_classes=3)
-    >>> model = model.fit(X, y)
-    >>> predictions = model.predict(X)  # Returns class labels, shape (n_samples, n_neurons)
+    >>> model = nmo.glm.ClassifierPopulationGLM(n_classes=3).fit(X, y)
+    >>> model.coef_.shape
+    (2, 2, 3)
+
+    **Predict Class Labels**
+
+    Get predicted class labels for each neuron:
```
Comment (gviejo) [inline src/nemos/glm/classifier_glm.py]:
```suggestion
    Get predicted class labels for each subject:
```

File: src/nemos/glm/classifier_glm.py [hunk 3]
```diff
@@ -908,14 +963,70 @@ class labels from the provided ``y``. If you set ``coef_`` and ``intercept_`` ma
...
+
+    Get predicted class labels for each subject:
+
+    >>> predictions = model.predict(X)
+    >>> predictions.shape
+    (6, 2)
+
+    **Predict Class Probabilities**
+
+    Get class probabilities for each neuron:
```
Comment (BalzaniEdoardo) [inline src/nemos/glm/classifier_glm.py]:
```suggestion
    Get class probabilities for each subject:
```
